cmd/main.go

```diff
@@ -242,8 +244,73 @@ func main() {
 		}
 	}
 
-	nimState := os.Getenv("NIM_STATE")
 	signalHandlerCtx := ctrl.SetupSignalHandler()
+	setupNim(mgr, signalHandlerCtx, kubeClient)
+
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
+		if kserveWithMeshEnabled {
+			if err = webhookservingv1.SetupServiceWebhookWithManager(mgr); err != nil {
+				setupLog.Error(err, "unable to create webhook", "webhook", "Knative Service")
+				os.Exit(1)
+			}
+		} else {
+			setupLog.Info("Skipping setup of Knative Service validating/mutating Webhook, " +
+				"because KServe Serverless setup seems to be disabled.")
+		}
+	}
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
+		if err = webhookservingv1beta1.SetupInferenceServiceWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "InferenceService")
+			os.Exit(1)
+		}
+	}
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
+		if err = webhookservingv1alpha1.SetupInferenceGraphWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "InferenceGraph")
+			os.Exit(1)
+		}
+	}
+
+	inferenceGraphCrdAvailable, igCrdErr := utils.IsCrdAvailable(
+		mgr.GetConfig(),
+		v1alpha1.SchemeGroupVersion.String(),
+		"InferenceGraph")
+	if igCrdErr != nil {
+		setupLog.Error(igCrdErr, "unable to check if InferenceGraph CRD is available", "controller", "InferenceGraph")
+		os.Exit(1)
+	} else if inferenceGraphCrdAvailable {
+		if err = servingcontroller.NewInferenceGraphReconciler(mgr).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "InferenceGraph")
+			os.Exit(1)
+		}
+	} else {
+		setupLog.Info("controller is turned off", "controller", "InferenceGraph")
+	}
+	// +kubebuilder:scaffold:builder
```

Comment:
is this expected to be added here?

maybe these scaffold instruction can be moved to the beginning of the file.

Reply:
It is not added. It has been always here at last.
GitHub diff is not smart enough to find that I extracted some code in the middle of `main()` to the `setupNim` func. So, in the diff it looks like I added a lot, when in reality I moved code around.

You may find a much better diff view here: https://app.semanticdiff.com/gh/opendatahub-io/odh-model-controller/pull/358/changes#cmd/main.go?ignore_comments=false

Reply:
BTW, we could move it to the top. It will later move to the middle, since `kubebuilder` is always pushing it down.

Reply:
that's fine.